evaluator/src/main/kotlin/Evaluator.kt

```diff
@@ -39,7 +39,8 @@ import org.ossreviewtoolkit.utils.scripting.ScriptRunner
 class Evaluator(
     ortResult: OrtResult = OrtResult.EMPTY,
     licenseInfoResolver: LicenseInfoResolver = OrtResult.EMPTY.createLicenseInfoResolver(),
-    licenseClassifications: LicenseClassifications = LicenseClassifications()
+    licenseClassifications: LicenseClassifications = LicenseClassifications(),
+    evaluationTime: Instant = Instant.now()
```

Comment:
I'm wondering whether we should somehow deduplicate this with the `startTime` in `run()`... it would somehow be nice if the start time written to the evaluator result would match the time the script was seeing.

Reply:
I thought about this as well and decided against it, because of the following
- measurement of execution time IMO becomes less clear from the code and duration is no more limited to the execution time of the `run()` function but involves the time from construction to the return of `run()`.
- introduces the assumption that the current time is passed to the constructor of the evaluator
- single value is used for multiple purposes, exec duration + parameter for the rules

What do you think?

Reply:
> * measurement of execution time IMO becomes less clear from the code and duration is no more limited to the execution time of the `run()` function but involves the time from construction to the return of `run()`.

As a side note, IIRC the *main* purpose of capturing `startTime` and `endTime` was *not* to capture the duration (otherwise we could have simply serialized the calculated duration), but to store the absolute time when then run was performed to e.g. ease debugging of network issues at that time. And only as a *bonus*, also the duration can be calculated.

> introduces the assumption that the current time is passed to the constructor of the evaluator

That coupling *might* indeed be a problem.

> single value is used for multiple purposes, exec duration + parameter for the rules

Using a single value for multiple purposes is not a problem per se, IMO, if the purposes are strongly related, as they are here.

But esp. for the case of testing I can see how you want to capture the correct `startTime` / `endTime` but set an arbitrary `evaluationTime`, so I'm fine with the code as-is.

However, please add these considerations to the commit message, so we can later still comprehend the design decision.

Reply:
The purpose of the `evaluationTime` should also be documented at the constructor argument so that you don't have to check the Git history to find out what it is used for.